pkg/metrics/real_time_test.go

```diff
@@ -148,3 +151,65 @@ func TestCalculateUsageNanoCores(t *testing.T) {
 	}
 
 }
+
+func TestFilterOutContainerNotInPod(t *testing.T) {
+
+	fake_container1 := "fake-container-name1"
+	fake_container2 := "fake-container-name2"
+	fake_container3 := "fake-infra-sidecar-container"
+	fake_container4 := "fake-container-name4"
+	podName := "pod-" + uuid.New().String()
+	podNamespace := "ns-" + uuid.New().String()
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: podNamespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name: fake_container1,
+				},
+				{
+					Name: fake_container2,
+				},
+				{
+					Name: fake_container4,
+				},
+			},
+		},
+	}
+
+	podStats := &realtimeMetricsExtensionPodStats{
+		Timestamp: 1234560000,
+		Containers: []containerStats{
+			{
+				Name: fake_container1,
+				CPU: cpuStats{
+					UsageCoreNanoSeconds: 2345678900,
+				},
+			},
+			{
+				Name: fake_container2,
+				CPU: cpuStats{
+					UsageCoreNanoSeconds: 1234567800,
+				},
+			},
+			{
```

Comment:
While the test case will pass but this will confuse the developer. As the realtime will get the metrics for all containers in this pod. It's impossible to have one pod runs different types of containers (k8s vs ACI).

Reply:
Understood. Added fake-container4 realtime metrics in test

Reply:
But `fake_container3` is not in the containers list

Reply:
Not sure I understand. Isn't the purpose of the function to filter out fake_container3 from containerStats as the container is not in the container list of pods?

Reply:
https://github.com/virtual-kubelet/azure-aci/blob/c88ad2ee2db41b678120da70253c5162a0c5dc54/pkg/metrics/real_time.go#L268

Reply:
It's the other way, the realtime extension can retrieve other containers (side cars) which are not in the pod.